misk/src/main/kotlin/misk/web/interceptors/RequestLogContextInterceptor.kt

```diff
@@ -0,0 +1,62 @@
+package misk.web.interceptors
+
+import misk.Action
+import misk.MiskCaller
+import misk.scope.ActionScoped
+import misk.web.NetworkChain
+import misk.web.NetworkInterceptor
+import misk.web.Response
+import org.slf4j.MDC
+import javax.inject.Inject
+import javax.inject.Singleton
+import javax.servlet.http.HttpServletRequest
+
+/**
+ * [RequestLogContextInterceptor] puts information about the current request into the
+ * logging MDC so it can be included in structured logs
+ */
+internal class RequestLogContextInterceptor private constructor(
+  private val action: Action,
+  private val currentCaller: ActionScoped<MiskCaller?>,
+  private val currentRequest: ActionScoped<HttpServletRequest>
+) : NetworkInterceptor {
+
+  override fun intercept(chain: NetworkChain): Response<*> {
+    val request = currentRequest.get()
+    return try {
+      MDC.put(MDC_ACTION, action.name)
+      MDC.put(MDC_CALLING_PRINCIPAL, currentCaller.get()?.principal ?: "unknown")
+      MDC.put(MDC_REMOTE_IP, request.remoteAddr)
+      MDC.put(MDC_REMOTE_PORT, request.remotePort.toString())
+      MDC.put(MDC_REQUEST_URI, request.requestURI)
```

Comment:
I could see somebody adding a new field here and forgetting to update the `allContextNames` below. Can we some kind of test that verifies that MDC is cleared of all keys?

Reply:
Yeah, this ends up being super tricky. Need to think about how to test it.

Reply:
maybe in a test just clear the MDC before, run the interceptor and verify nothing is in the MDC after?

Reply:
yes but that's tricky because there needs to be something on the other side of the interceptor, plus a bunch of fakes for the action scoped, etc. it'll take a bunch of code